gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/NewestKRetentionPolicy.java

```diff
@@ -20,16 +21,33 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(NewestKRetentionPolicy.class);
 
+  /**
+   * @deprecated use {@link #NEWEST_K_VERSIONS_RETAINED_KEY}
+   */
+  @Deprecated
   public static final String VERSIONS_RETAINED_KEY = DatasetCleaner.CONFIGURATION_KEY_PREFIX +
       "versions.retained";
+
+  public static final String NEWEST_K_VERSIONS_RETAINED_KEY = DatasetCleaner.CONFIGURATION_KEY_PREFIX +
+      "newestK.versions.retained";
+
   public static final String VERSIONS_RETAINED_DEFAULT = Integer.toString(2);
 
   private final int versionsRetained;
 
-  public NewestKRetentionPolicy(Properties props) {
-    this.versionsRetained = Integer.parseInt(props.getProperty(VERSIONS_RETAINED_KEY, VERSIONS_RETAINED_DEFAULT));
+  public NewestKRetentionPolicy(String versionsRetained) {
+    this.versionsRetained = Integer.parseInt(versionsRetained);
     LOGGER.info(String.format("%s will retain %d versions of each dataset.",
-        NewestKRetentionPolicy.class.getCanonicalName(), this.versionsRetained));
+        NewestKRetentionPolicy.class.getName(), this.versionsRetained));
+  }
+
+  public NewestKRetentionPolicy(Properties props) {
+    this(props.getProperty(VERSIONS_RETAINED_KEY, VERSIONS_RETAINED_DEFAULT));
```

Comment:
Retained the old key for backward compatibility. Added code to check new key if old key is not found.